chore: standardize count metric descriptions on "Number of" (#3314)
NVIDIA/bare-metal-manager-core: docs/observability/core_metrics.md

<tr><td>carbide_hosts_unhealthy_by_probe_id_count</td><td>gauge</td><td>Number of objects which reported a certain Health Probe Alert</td></tr>
<tr><td>carbide_hosts_usable_count</td><td>gauge</td><td>Number of remaining hosts in the NICo deployment which are available for immediate instance creation</td></tr>
<tr><td>carbide_hosts_with_bios_password_set</td><td>gauge</td><td>Number of hosts in the system that have their BIOS password set.</td></tr>
<tr><td>carbide_ib_monitor_fabrics_count</td><td>gauge</td><td>Number of InfiniBand fabrics that are monitored</td></tr>
<tr><td>carbide_ib_monitor_machine_ib_status_updates_count</td><td>gauge</td><td>Number of machines whose InfiniBand status observation was updated</td></tr>
<tr><td>carbide_ib_monitor_machines_with_missing_pkeys_count</td><td>gauge</td><td>Number of machines where at least one port is not assigned to the expected pkey on UFM</td></tr>
<tr><td>carbide_ib_monitor_machines_with_unexpected_pkeys_count</td><td>gauge</td><td>Number of machines where at least one port is assigned to an unexpected pkey on UFM</td></tr>

> <tr><td>carbide_ib_monitor_machines_with_unknown_pkeys_count</td><td>gauge</td><td>Number of machines where at least one port is assigned to a pkey value that is not associated with any partition ID</td></tr>

Reviewer: ```suggestion
<tr><td>carbide_ib_monitor_machines_with_unknown_pkeys_count</td><td>gauge</td><td>Number of machines where at least one port is assigned to a pkey value not associated with any partition ID</td></tr>
```

Author: @polarweasel Done, thank you!

<tr><td>carbide_ib_partitions_enqueuer_iteration_latency_milliseconds</td><td>histogram</td><td>The overall time it took to enqueue state handling tasks for all carbide_ib_partitions in the system</td></tr>
<tr><td>carbide_ib_partitions_iteration_latency_milliseconds</td><td>histogram</td><td>The elapsed time in the last state processor iteration to handle objects of type carbide_ib_partitions</td></tr>
<tr><td>carbide_ib_partitions_object_tasks_enqueued_total</td><td>counter</td><td>Number of object handling tasks that have been freshly enqueued for objects of type carbide_ib_partitions</td></tr>
<tr><td>carbide_ib_partitions_total</td><td>gauge</td><td>Number of carbide_ib_partitions in the system</td></tr>
<tr><td>carbide_log_events_total</td><td>counter</td><td>Number of log events emitted, by level and component. The always-on log-volume and error-rate signal for every binary.</td></tr>
<tr><td>carbide_machine_reboot_duration_seconds</td><td>histogram</td><td>Time taken for machine/host to reboot in seconds</td></tr>

> <tr><td>carbide_machine_updates_started_count</td><td>gauge</td><td>Number of machines in the system that are in the process of updating.</td></tr>

Reviewer: ```suggestion
<tr><td>carbide_machine_updates_started_count</td><td>gauge</td><td>Number of machines in the system in the process of updating</td></tr>
```

Author: @polarweasel Done, thank you!

<tr><td>carbide_machine_validation_completed</td><td>gauge</td><td>Number of machine validation runs that have completed successfully</td></tr>
<tr><td>carbide_machine_validation_failed</td><td>gauge</td><td>Number of machine validation runs that have failed</td></tr>

> <tr><td>carbide_machine_validation_in_progress</td><td>gauge</td><td>Number of machine validation runs that are in progress</td></tr>

Reviewer: "That are" and "that have" are more of those phrases that can probably disappear in a lot of cases in a table like this...

```suggestion
<tr><td>carbide_machine_validation_completed</td><td>gauge</td><td>Number of successfully completed machine validation runs</td></tr>
<tr><td>carbide_machine_validation_failed</td><td>gauge</td><td>Number of failed machine validation runs</td></tr>
<tr><td>carbide_machine_validation_in_progress</td><td>gauge</td><td>Number of machine validation runs in progress</td></tr>
```

Author: @polarweasel Done, thank you!

<tr><td>carbide_measured_boot_machines_per_bundle_state_total</td><td>gauge</td><td>Number of machines in each measured boot bundle state.</td></tr>
<tr><td>carbide_measured_boot_machines_per_machine_state_total</td><td>gauge</td><td>Number of machines in each measured boot machine state.</td></tr>
<tr><td>carbide_measured_boot_machines_total</td><td>gauge</td><td>Number of machines reporting measurements.</td></tr>
<tr><td>carbide_measured_boot_profiles_total</td><td>gauge</td><td>Number of measured boot profiles.</td></tr>
<tr><td>carbide_network_segments_enqueuer_iteration_latency_milliseconds</td><td>histogram</td><td>The overall time it took to enqueue state handling tasks for all carbide_network_segments in the system</td></tr>
<tr><td>carbide_network_segments_handler_latency_in_state_milliseconds</td><td>histogram</td><td>The amount of time it took to invoke the state handler for objects of type carbide_network_segments in a certain state</td></tr>
<tr><td>carbide_network_segments_iteration_latency_milliseconds</td><td>histogram</td><td>The elapsed time in the last state processor iteration to handle objects of type carbide_network_segments</td></tr>

> <tr><td>carbide_network_segments_object_tasks_completed_total</td><td>counter</td><td>Number of object handling tasks that have been completed for objects of type carbide_network_segments</td></tr>

Reviewer: "That have been" can probably disappear in most places in this summary table :)
```suggestion
<tr><td>carbide_network_segments_object_tasks_completed_total</td><td>counter</td><td>Number of object handling tasks completed for objects of type carbide_network_segments</td></tr>
```

Author: @polarweasel Applied across the whole templated family -- enqueued/dispatched/completed/requeued for every object type. Done, thank you!

<tr><td>carbide_network_segments_with_state_handling_errors_per_state</td><td>gauge</td><td>Number of state-handling errors for carbide_network_segments in a given state</td></tr>
<tr><td>carbide_nvlink_partition_monitor_machine_status_updates_count</td><td>gauge</td><td>Number of machines whose NVLink status observation was updated</td></tr>
<tr><td>carbide_nvlink_partition_monitor_nmxc_changes_applied_total</td><td>counter</td><td>Number of changes requested to NMX-C</td></tr>
<tr><td>carbide_nvlink_partition_monitor_num_logical_partitions</td><td>gauge</td><td>Number of logical partitions that were monitored</td></tr>
<tr><td>carbide_nvlink_partition_monitor_num_physical_partitions</td><td>gauge</td><td>Number of physical partitions that were monitored</td></tr>
<tr><td>carbide_nvlink_partition_monitor_nvlink_info_mismatches</td><td>gauge</td><td>Number of NVLink GPU partition ID mismatches between DB and NMX-C</td></tr>
<tr><td>carbide_nvlink_partition_monitor_stale_partitions_deleted</td><td>gauge</td><td>Number of stale partitions deleted from DB (not found in NMX-C)</td></tr>

> <tr><td>carbide_pending_dpu_nic_firmware_update_count</td><td>gauge</td><td>Number of machines in the system that need a firmware update.</td></tr>

Reviewer: Some of these end in periods, but most don't. It's an annoyance, but not really a thing that warrants urgent fixing.
```suggestion
<tr><td>carbide_pending_dpu_nic_firmware_update_count</td><td>gauge</td><td>Number of machines in the system that need a firmware update</td></tr>
```

Author: @polarweasel Dropped the periods across this family while applying the DPU/NIC callout. Done, thank you!

<tr><td>carbide_power_shelves_object_tasks_enqueued_total</td><td>counter</td><td>Number of object handling tasks that have been freshly enqueued for objects of type carbide_power_shelves</td></tr>
<tr><td>carbide_power_shelves_total</td><td>gauge</td><td>Number of carbide_power_shelves in the system</td></tr>
<tr><td>carbide_preingestion_total</td><td>gauge</td><td>Number of known machines currently being evaluated prior to ingestion</td></tr>
<tr><td>carbide_preingestion_waiting_download</td><td>gauge</td><td>Number of machines that are waiting for firmware downloads on other machines to complete before doing their own</td></tr>
<tr><td>carbide_preingestion_waiting_installation</td><td>gauge</td><td>Number of machines which have had firmware uploaded to them and are currently in the process of installing that firmware</td></tr>
<tr><td>carbide_racks_enqueuer_iteration_latency_milliseconds</td><td>histogram</td><td>The overall time it took to enqueue state handling tasks for all carbide_racks in the system</td></tr>

> <tr><td>carbide_racks_health_overrides_count</td><td>gauge</td><td>Number of health overrides that are configured in the site</td></tr>

Reviewer: Things like this, it's best to eliminate words if we can
```suggestion
<tr><td>carbide_racks_health_overrides_count</td><td>gauge</td><td>Number of health overrides configured in the site</td></tr>
```

Author: @polarweasel Applied at the source in `health-metrics`, so all four per-object-type rows pick it up. Done, thank you!

<tr><td>carbide_api_ready</td><td>gauge</td><td>Whether the NICo API is running</td></tr>
<tr><td>carbide_api_tls_cert_refreshes_total</td><td>counter</td><td>Number of TLS acceptor refreshes performed by the API listener</td></tr>
<tr><td>carbide_api_tls_connection_attempted_total</td><td>counter</td><td>Number of inbound TLS connection attempts</td></tr>
<tr><td>carbide_api_tls_connection_success_total</td><td>counter</td><td>Number of successful TLS connections</td></tr>
<tr><td>carbide_api_tracing_spans_open</td><td>gauge</td><td>Number of open logging/tracing spans</td></tr>
<tr><td>carbide_api_vault_request_duration_milliseconds</td><td>histogram</td><td>the duration of outbound vault requests, in milliseconds</td></tr>
<tr><td>carbide_api_vault_requests_attempted_total</td><td>counter</td><td>Number of vault requests attempted</td></tr>
<tr><td>carbide_api_vault_requests_failed_total</td><td>counter</td><td>Number of vault requests that failed</td></tr>
<tr><td>carbide_api_vault_requests_succeeded_total</td><td>counter</td><td>Number of vault requests that succeeded</td></tr>

> <tr><td>carbide_api_vault_token_time_until_refresh_seconds</td><td>gauge</td><td>The amount of time, in seconds, until the vault token is required to be refreshed</td></tr>

Author: Done -- `Vault` is capitalized across the client metrics (`Number of attempted/successful/failed Vault requests`), the duration histogram now opens with `Duration of outbound Vault requests`, and the doc regenerates to match. PTAL. Resolve if good, thanks!

<tr><td>carbide_hosts_health_status_count</td><td>gauge</td><td>Number of managed hosts in the system that have reported either a healthy or not healthy status - based on the presence of health probe alerts</td></tr>
<tr><td>carbide_hosts_in_use_count</td><td>gauge</td><td>Number of hosts that are actively used by tenants as instances in the NICo deployment</td></tr>
<tr><td>carbide_hosts_unhealthy_by_classification_count</td><td>gauge</td><td>Number of objects marked with a certain classification due to being unhealthy</td></tr>
<tr><td>carbide_hosts_unhealthy_by_probe_id_count</td><td>gauge</td><td>Number of objects which reported a certain Health Probe Alert</td></tr>
<tr><td>carbide_hosts_usable_count</td><td>gauge</td><td>Number of remaining hosts in the NICo deployment which are available for immediate instance creation</td></tr>
<tr><td>carbide_hosts_with_bios_password_set</td><td>gauge</td><td>Number of hosts in the system that have their BIOS password set.</td></tr>
<tr><td>carbide_ib_monitor_fabrics_count</td><td>gauge</td><td>Number of monitored InfiniBand fabrics</td></tr>

> <tr><td>carbide_ib_monitor_machine_ib_status_updates_count</td><td>gauge</td><td>Number of Machines whose InfiniBand status observation was updated</td></tr>

Author: Declining -- @polarweasel explicitly asked to keep the capital: `Machines` is the entity name in this codebase and his suggestion on this exact line restored it. Deferring to the tech writer on published wording. Thanks for checking!

<tr><td>carbide_hosts_health_status_count</td><td>gauge</td><td>Number of Managed Hosts in the system that have reported either a healthy or not healthy status - based on the presence of health probe alerts</td></tr>
<tr><td>carbide_hosts_in_use_count</td><td>gauge</td><td>Number of hosts that are actively used by tenants as instances in the NICo deployment</td></tr>
<tr><td>carbide_hosts_unhealthy_by_classification_count</td><td>gauge</td><td>Number of objects which are marked with a certain classification due to being unhealthy</td></tr>
<tr><td>carbide_hosts_unhealthy_by_probe_id_count</td><td>gauge</td><td>Number of objects which reported a certain Health Probe Alert</td></tr>
<tr><td>carbide_hosts_usable_count</td><td>gauge</td><td>Number of remaining hosts in the NICo deployment which are available for immediate instance creation</td></tr>
<tr><td>carbide_hosts_with_bios_password_set</td><td>gauge</td><td>Number of Hosts in the system that have their BIOS password set.</td></tr>
<tr><td>carbide_ib_monitor_fabrics_count</td><td>gauge</td><td>Number of InfiniBand fabrics that are monitored</td></tr>

> <tr><td>carbide_ib_monitor_machine_ib_status_updates_count</td><td>gauge</td><td>Number of Machines where the infiniband_status_observation got updated</td></tr>

Author: Reworded both at the source -- `Number of machines whose InfiniBand status observation was updated` / `Number of machines whose NVLink status observation was updated` -- and the doc rows match. PTAL. Resolve if good, thanks!
